### PR TITLE
docs: Add documentation for `ActsFatras`

### PR DESCRIPTION
**Pull request by andiwand**
related issues
- #2796

### CODE REVIEW

File: docs/fatras/fatras.md [hunk 1]
```diff
@@ -1,8 +1,35 @@
...
+A dedicated description of the propagation can be found [here](propagation_impl).
 :::
 
+Hits on sensitive surfaces are recorded and tagged with the particle passing through the surface.
+This information can afterwards be used by a digitization algorithm to mimic a detector response.
+
+ACTS Fatras is fully capable of handling secondary particles even tho the current set of interactions is not utilizing this to a big extent.
+The simulation will propagate each particle until it reaches the end of the detector or until a specified path length is reached.
+Afterwards the first secondary particle that might have been generated in the process will be propagated.
+This continues until all particles are transported.
```
Comment (AJPfleger) [inline docs/fatras/fatras.md:18]:
```suggestion
Hits on sensitive surfaces are recorded and tagged with the particle passing through the surface.
Later, this hit information can be used by a digitization algorithm to mimic a detector response.

ACTS Fatras is fully capable of handling secondary particles even though the current set of interactions is not utilizing this to a big extent.
The simulation will propagate each particle until it reaches the end of the detector or until a specified path length is reached.
Afterwards, the first secondary particle that might have been generated in the process will be propagated.
This continues until all particles are transported.
```

File: docs/fatras/fatras.md [hunk 2]
```diff
@@ -1,8 +1,35 @@
...
+ - Bethe-Heitler energy loss for electrons (and positrons) {class}`ActsFatras::BetheHeitler`
+ - Photon conversion for pair production {class}`ActsFatras::PhotonConversion`
+ - Scattering for charged particles {class}`ActsFatras::GenericScattering`
+
+These interactions are meant to be physical accurate within their boundaries but are far from sufficient for a full simulation.
+
+## Use-cases
+
+The primary use-case for ACTS Fatras is early algorithm validation.
+It is fast enough to simulate thousands of small to medium events (about 100 tracks) on consumer hardware in a few seconds which provides a quick turnaround time for development.
```
Comment (AJPfleger) [inline docs/fatras/fatras.md:33]:
```suggestion
The primary use-case for ACTS Fatras is early algorithm validation.
It is fast enough to simulate thousands of small to medium events (about 100 tracks) on consumer hardware in a few seconds.
This provides a quick turnaround time for development.
```

**Review by AJPfleger — CHANGES_REQUESTED** [review]

**Review by AJPfleger — APPROVED** [review]

File: docs/fatras/fatras.md [hunk 3]
```diff
@@ -1,8 +1,36 @@
...
+This continues until all particles are transported.
+
 ## Supported interactions
-## Use-cases & counter indications (when to use G4)
+
+Fatras implements a few interactions
+ - Bethe-Bloch energy loss for charged particles {class}`ActsFatras::BetheBloch`
+ - Bethe-Heitler energy loss for electrons (and positrons) {class}`ActsFatras::BetheHeitler`
+ - Photon conversion for pair production {class}`ActsFatras::PhotonConversion`
+ - Scattering for charged particles {class}`ActsFatras::GenericScattering`
```
Comment (andiwand) [inline docs/fatras/fatras.md:26]:
```suggestion
 - Bethe-Bloch energy loss for charged particles (see `ActsFatras::BetheBloch`)
 - Bethe-Heitler energy loss for electrons (and positrons) (see `ActsFatras::BetheHeitler`)
 - Photon conversion for pair production (see `ActsFatras::PhotonConversion`)
 - Scattering for charged particles (see `ActsFatras::GenericScattering`)
```

**Review by AJPfleger — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [831bf73c136d94f264bc6bd9bc4efd8bca455938]
### Build job with this PR failed!
Please investigate the [build job](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34708236) for the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/6613851)!

**Pull request merged**